eth/stagedsync/stage_log_index.go

```diff
@@ -175,6 +177,9 @@ func promoteLogIndex(logPrefix string, tx kv.RwTx, start uint64, endBlock uint64
 		}
 
 		for _, l := range ll {
+			if cfg.noPruneContracts != nil && !cfg.noPruneContracts[l.Address] && l.BlockNumber < pruneBlock {
```

Comment:
for performance maybe better place `l.BlockNumber < pruneBlock` before map check